source/common/network/listener_filter_manager_impl_base.cc

```diff
@@ -0,0 +1,50 @@
+#include "source/common/network/listener_filter_manager_impl_base.h"
+
+namespace Envoy {
+namespace Network {
+
+class GenericListenerFilter : public Network::ListenerFilter {
```

Comment:
There is GenericListenerFilter defined here: https://github.com/envoyproxy/envoy/blob/20676d718086b5a91b991f9dad45e029493f2ab9/source/extensions/listener_managers/listener_manager/active_tcp_socket.h#L42

Is there a reason to define a separate copy in the source/common/network?

Reply:
I meant to share code between QUIC and TCP, but later the interface diverged. Instead of the listener manager, I added a listern filter template for them.